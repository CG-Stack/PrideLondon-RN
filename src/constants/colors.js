export const bgColor = "#808080";
export const cardBgColor = "#FFFFFF";
export const imageBgColor = "#C1C1C1";
export const headerBgColor = "#A5A5A5";
export const priceTagBgColor = "#989898";

export const textColor = "#272727";

export const dialogBackdropColor = "rgba(0, 0, 0, 0.7)";
export const dialogBgColor = "#FFFFFF";
export const dialogApplyButtonBgColor = "rgb(51, 51, 51)";
export const dialogApplyButtonTextColor = "#FFFFFF";
export const dialogHeaderDividerColor = "#F2F2F2";

// FILTER HEADER

export const interestButtonBgColor = "rgb(32, 33, 90)";
export const interestButtonTextColor = "rgb(44, 218, 157)";
export const filterButtonBorderColor = "rgb(44, 218, 157)";
export const filterButtonTextColor = "rgb(44, 218, 157)";

// EVENT LIST

export const eventListBgColor = "rgb(226, 226, 226)";
export const eventListHeaderBgColor = "rgb(216, 216, 216)";
export const eventListHeaderColor = "rgb(51, 51, 51)";
export const lightNavyBlue = "rgb(45, 47, 127)";
export const lightGrey = "rgb(243, 243, 243)";
export const white = "rgb(255, 255, 255)";
export const eventCardShadow = "rgba(0, 0, 0, 0.2)";
export const sectionHeaderShadow = "rgba(0, 0, 0, 0.14)";
export const eucalyptusGreen = "rgb(44, 218, 157)";
export const darkBlueGreyTwo = "rgb(32, 33, 90)";

// EVENT DETAILS
export const eventDetailsBgColor = "#FFF";
export const eventDetailsHeaderBgColor = "#F2F2F2";
export const eventCategoryLabelBgColor = "#D8D8D8";
export const eventDetailsBuyButtonColor = "#D8D8D8";
export const eventDetailsBuyButtonTextColor = "#FFFFFF";
