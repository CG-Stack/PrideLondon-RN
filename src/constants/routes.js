export const EVENT_DETAILS = "EVENT_DETAILS";
export const EVENT_LIST = "EVENT_LIST";
export const FEATURED_EVENT_LIST = "FEATURED_EVENT_LIST";
export const HOME = "HOME";
export const EVENT_CATEGORY_FILTER = "EVENT_CATEGORY_FILTER";
