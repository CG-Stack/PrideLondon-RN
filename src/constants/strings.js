export default {
  venuedetailsGenderNeutralToilets: "Gender neutral toilets"
};
