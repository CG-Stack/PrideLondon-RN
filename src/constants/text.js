export default {
  tabEvents: "Events",
  filterByInterest: "I'm interested in...",
  eventDetailsPrice: "From: Â£",
  eventDetailsAccessibility: "Accessibility",
  eventDetailsAccessibilityDetails: "Accessibility Details",
  eventDetailsContact: "Contact",
  eventDetailsBuyButton: "Buy Tickets",
  eventDetailsGenderNeutralToilets: "Gender neutral toilets"
};
