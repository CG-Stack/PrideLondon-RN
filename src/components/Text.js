// @flow
import React from "react";
import { Text as RnText, StyleSheet } from "react-native";
import type { TextProps } from "react-native/Libraries/Text/TextProps";
import Markdown from "react-native-easy-markdown";
import { mergeDeepRight } from "ramda";
import { blackColor, lightNavyBlueColor } from "../constants/colors";

export type TextType = "h1" | "h2" | "h3" | "h4" | "text" | "small" | "price";
export type TextColor = "lightNavyBlueColor" | "blackColor";

type Props = {
  type: TextType,
  color: TextColor,
  markdown: boolean,
  ...TextProps
};

class Text extends React.PureComponent<Props> {
  static defaultProps = {
    type: "text",
    markdown: false,
    color: "blackColor"
  };

  render() {
    const { color, type, markdown, style, ...otherProps } = this.props;
    return markdown ? (
      <Markdown
        style={style}
        markdownStyles={mergeDeepRight(textStyles, markdownStyles)}
        {...otherProps}
      />
    ) : (
      <RnText
        style={[styles[type], color && styles[color], style]}
        {...otherProps}
      />
    );
  }
}

const textStyles = {
  h1: {
    fontFamily: "Poppins-Bold",
    fontSize: 24,
    lineHeight: 28,
    includeFontPadding: false
  },
  h2: {
    fontFamily: "Poppins-SemiBold",
    fontSize: 18,
    lineHeight: 24,
    includeFontPadding: false
  },
  h3: {
    fontFamily: "Poppins-SemiBold",
    fontSize: 16,
    lineHeight: 20,
    includeFontPadding: false
  },
  h4: {
    fontFamily: "Roboto-Medium",
    fontSize: 16,
    lineHeight: 24
  },
  text: {
    fontFamily: "Roboto",
    fontSize: 16,
    lineHeight: 24
  },
  small: {
    fontFamily: "Roboto",
    fontSize: 14,
    lineHeight: 20
  },
  price: {
    fontFamily: "Roboto-Bold",
    fontSize: 14,
    lineHeight: 20
  },
  blackColor: {
    color: blackColor
  },
  lightNavyBlueColor: {
    color: lightNavyBlueColor
  }
};

const markdownStyles = {
  // "u" - underline is listed in react-native-easy-markdown
  // but doesn't exist in the markdown spec so should be rendered bold
  u: {
    fontWeight: "bold"
  },
  text: {
    color: blackColor
  }
};

const styles = StyleSheet.create(textStyles);

export default Text;
