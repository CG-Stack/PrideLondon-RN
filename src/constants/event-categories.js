// @flow
// Ideally the keys here should be consts, but currently these are setup in
// contentful only as the label, hence we need to use this for the key too.
// If we move to realise categories as its own content type then this will
// go away and we will fetch this data.
import * as colors from "./colors";
import type { EventCategoryMap } from "../data/event-deprecated";

const categories: EventCategoryMap = {
  "en-GB": {
    "Cabaret and Variety": {
      label: "Cabaret and Variety",
      color: colors.coralColor,
      contrast: false
    },
    Community: {
      label: "Community",
      color: colors.brightLightBlueColor,
      contrast: true
    },
    "Talks and Debates": {
      label: "Talks and Debates",
      color: colors.darkSkyBlueColor,
      contrast: false
    },
    "Film and Screenings": {
      label: "Film and Screenings",
      color: colors.lightTealColor,
      contrast: true
    },
    "Plays and Theatre": {
      label: "Plays and Theatre",
      color: colors.warmPinkColor,
      contrast: false
    },
    "Social and Networking": {
      label: "Social and Networking",
      color: colors.turquoiseBlueColor,
      contrast: true
    },
    Nightlife: {
      label: "Nightlife",
      color: colors.yellowColor,
      contrast: true
    },
    "Exhibition and Tours": {
      label: "Exhibition and Tours",
      color: colors.brightPurpleColor,
      contrast: false
    },
    "Sports and Activities": {
      label: "Sports and Activities",
      color: colors.vomitYellowColor,
      contrast: true
    },
    Health: {
      label: "Health",
      color: colors.bubblegumPinkColor,
      contrast: true
    },
    Music: {
      label: "Music",
      color: colors.cornflowerBlueColor,
      contrast: false
    }
  }
};

export default categories;
