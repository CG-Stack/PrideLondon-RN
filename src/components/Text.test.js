// @flow
import React from "react";
import { shallow } from "enzyme";
import Markdown from "react-native-easy-markdown";
import Text from "./Text";
import { lightNavyBlueColor } from "../constants/colors";

it("renders correctly", () => {
  const output = shallow(<Text>Some text</Text>);
  expect(output).toMatchSnapshot();
});

it("renders text based on type", () => {
  const output = shallow(<Text type="h1">Some text</Text>);

  expect(output.props().style[0]).toEqual(
    expect.objectContaining({ fontFamily: "Poppins-Bold" })
  );
});

it("renders markdown component based on flag", () => {
  const output = shallow(<Text markdown>**Some markdown**</Text>);

  expect(output).toMatchSnapshot();
});

it("does not render markdown images", () => {
  const output = shallow(
    <Text markdown>![Test](https://placehold.it/320x320.png)</Text>
  );
  const markdown = output.find(Markdown).shallow();
  console.log(markdown);
  expect(markdown).toMatchSnapshot();
});

it("passes custom styles down to component", () => {
  const style = { fontSize: 300 };
  const output = shallow(<Text style={style}>Some text</Text>);

  expect(output.props().style).toContainEqual(style);
});

it("renders text in blue when color is set", () => {
  const style = { color: lightNavyBlueColor };
  const output = shallow(<Text color="lightNavyBlueColor">Some text</Text>);

  expect(output.props().style).toContainEqual(style);
});
