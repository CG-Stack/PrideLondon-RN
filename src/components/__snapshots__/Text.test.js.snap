// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onLayout={[Function]}
  style={
    Array [
      Object {
        "fontFamily": "Roboto",
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {},
    ]
  }
>
  Some text
</Text>
`;

exports[`renders markdown component based on flag 1`] = `
<Markdown
  allowFontScaling={true}
  debug={false}
  markdownStyles={
    Object {
      "h1": Object {
        "fontFamily": "Poppins-Bold",
        "fontSize": 24,
        "includeFontPadding": false,
        "lineHeight": 28,
      },
      "h2": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 18,
        "includeFontPadding": false,
        "lineHeight": 24,
      },
      "h3": Object {
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 16,
        "includeFontPadding": false,
        "lineHeight": 20,
      },
      "h4": Object {
        "fontFamily": "Roboto-Medium",
        "fontSize": 16,
        "lineHeight": 24,
      },
      "price": Object {
        "fontFamily": "Roboto-Bold",
        "fontSize": 14,
        "lineHeight": 20,
      },
      "small": Object {
        "fontFamily": "Roboto",
        "fontSize": 14,
        "lineHeight": 20,
      },
      "text": Object {
        "color": "#111111",
      },
      "u": Object {
        "fontWeight": "bold",
      },
    }
  }
  onLayout={[Function]}
  parseInline={false}
  style={Object {}}
  useDefaultStyles={true}
>
  **Some markdown**
</Markdown>
`;
